### PR TITLE
Switch snekfetch to superagent.

### LINKED ISSUE #2581: [Ratelimits] Snekfetch 4.x breaks ratelimits
**Please describe the problem you are having in as much detail as possible:**

After https://github.com/discordjs/discord.js/commit/46af6e3396e94f795c7f539035a36e733139be07 landed, 429 HTTP errors aren't handled as well as before that commit (for example, https://github.com/discordjs/discord.js/commit/c8ecbe6585a615e77deb5f04bf64a9675e73fb41).

I have tested in two bots, my production bot, with commit https://github.com/discordjs/discord.js/commit/6cde6deceee975d76ea3abf40e87bd6ab04655f7, which handled ratelimits correctly, and my test bot, with commit https://github.com/discordjs/discord.js/commit/5071477172053ef9d274aaab16cec0cfbd25b860, which threw a lot of 429 errors to console:

image

Furthermore, I have checked `this.client.rest.handlers`, and got the following:

```javascript
{ '/gateway':
   RequestHandler {
     manager: [RESTManager],
     client: [Smii],
     handle: [Function: bound sequential],
     limit: NaN,
     resetTime: NaN,
     remaining: NaN,
     queue: [],
     busy: false },
  '/oauth2/applications/@​me':
   RequestHandler {
     manager: [RESTManager],
     client: [Smii],
     handle: [Function: bound sequential],
     limit: NaN,
     resetTime: NaN,
     remaining: NaN,
     queue: [],
     busy: false },
  '/channels/339948984351326208/typing':
   RequestHandler {
     manager: [RESTManager],
     client: [Smii],
     handle: [Function: bound sequential],
     limit: 5,
     resetTime: NaN,
     remaining: 4,
     queue: [],
     busy: true },
  '/channels/339948984351326208/messages':
   RequestHandler {
     manager: [RESTManager],
     client: [Smii],
     handle: [Function: bound sequential],
     limit: 5,
     resetTime: NaN,
     remaining: 4,
     queue: [],
     busy: false },
  '/channels/339948984351326208/messages/:id':
   RequestHandler {
     manager: [RESTManager],
     client: [Smii],
     handle: [Function: bound sequential],
     limit: 5,
     resetTime: NaN,
     remaining: 0,
     queue: [],
     busy: false } }
```

**Downgrading to https://github.com/discordjs/discord.js/commit/c8ecbe6585a615e77deb5f04bf64a9675e73fb41 fixes the issue!**

**Include a reproducible code sample here, if possible:**

Install https://github.com/discordjs/discord.js/commit/46af6e3396e94f795c7f539035a36e733139be07 and make your bot send 6 messages in less than a second (to fill the bucket and make it set busy true).

**Further details:**

- discord.js version: master
- node.js version: 10.3.0
- Operating system: Ubuntu 16.04.3 LTS (Xenial Xerus)
- Priority this issue should have – please be realistic and elaborate if possible: High, I'm not the only one who reproduces this issue, several developers with massive bots have experienced this issue and they're currently getting many errors.

- [ ] I found this issue while running code on a __user account__
- [x] I have also tested the issue on latest master, commit hash: https://github.com/discordjs/discord.js/commit/5071477172053ef9d274aaab16cec0cfbd25b860

### PR DESCRIPTION
**Pull request by dragonfire535**
**Please describe the changes this PR makes and why it should be merged:**
While node-fetch is another option, superagent was what we used before, has a close API to snekfetch (only required minor changes to work), and personal bias of course. snekfetch is good, but even Gus himself said we should switch away, and this is one of many ways to do so.

fixes #2581 
closes #2582 
closes #2583 
closes #2587 
closes #2589 
closes #a bigger door than gus'

**Semantic versioning classification:**  
- [ ] This PR changes the library's interface (methods or parameters added)
  - [ ] This PR includes breaking changes (methods removed or renamed, parameters moved or removed)
- [ ] This PR **only** includes non-code changes, like changes to documentation, README, etc.

### CODE REVIEW

**Comment by devsnek** [comment]
-1, we should use node-fetch

**Comment by YoussefElshemi** [comment]
have you tested ratelimits with this?

**Comment by dragonfire535** [comment]
@YoussefElshemi yes, and they seem to be working as normal.

File: src/rest/APIRequest.js [hunk 1]
```diff
@@ -1,10 +1,7 @@
 const querystring = require('querystring');
-const snekfetch = require('snekfetch');
-const https = require('https');
+const superagent = require('superagent');
 const { browser, UserAgent } = require('../util/Constants');
 
-if (https.Agent) var agent = new https.Agent({ keepAlive: true });
```
Comment (devsnek) [inline src/rest/APIRequest.js]: this is pretty sad :(

Comment (dragonfire535) [reply src/rest/APIRequest.js]: I couldn't find any way to do it via superagent, so I assume no support.

**Pull request closed without merging**